src/main/kotlin/com/sourcegraph/cody/initialization/TrialEndedNotification.kt

```diff
@@ -39,6 +43,9 @@ class TrialEndedNotification(val disposable: Disposable) :
             disposable.dispose()
             notification.expire()
           }
+
+          @Suppress("MissingRecentApi")
+          override fun getActionUpdateThread(): ActionUpdateThread = ActionUpdateThread.BGT
```

Comment:
IIUC this is the other case (next to `DumbAwareAction`) where we need to define `getActionUpdateThread`, right? 

1. can we have a similar solution - `NotificationBGTAction`? 
2. can you add a CI check that verifies that we DO not use `NotificationAction` or `DumbAwareAction` in any other place than the corresponding abstract classes?

Reply:
Turns out for `NotificationAction` it is not really needed, I will revert this.
In fact it looks it is needed only on classes which overrides `update` method, so I will do a few more reverts.

Reply:
As for CI it's possible to run IJ inspection but I do not see an easy way to verify this in CI.